pkgs/by-name/ld/ld64/package.nix

```diff
@@ -84,84 +49,90 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
-  xcodeHash = "sha256-+j7Ed/6aD46SJnr3DWPfWuYWylb2FNJRPmWsUVxZJHM=";
-
-  postUnpack = ''
-    # Verify that the Xcode project has not changed unexpectedly.
-    hashType=$(echo $xcodeHash | cut -d- -f1)
-    expectedHash=$(echo $xcodeHash | cut -d- -f2)
-    hash=$(openssl "$hashType" -binary "$sourceRoot/ld64.xcodeproj/project.pbxproj" | base64)
-
-    if [ "$hash" != "$expectedHash" ]; then
-      echo 'error: hash mismatch in ld64.xcodeproj/project.pbxproj'
-      echo "        specified: $xcodeHash"
-      echo "           got:    $hashType-$hash"
-      echo
-      echo 'Upstream Xcode project has changed. Update `meson.build` with any changes, then update `xcodeHash`.'
-      echo 'Use `nix-hash --flat --sri --type sha256 ld64.xcodeproj/project.pbxproj` to regenerate it.'
-      exit 1
-    fi
+  patches =
+    [
+      # These patches are vendored from https://github.com/reckenrode/ld64/tree/ld64-951.9-nixpkgs.
+      # See their comments for more on what they do.
+      ./patches/0003-Always-use-write-instead-of-mmap.patch
+      ./patches/0005-Inline-missing-definitions-instead-of-using-private-.patch
+      ./patches/0006-Removed-unused-Blob-clone-method.patch
+      ./patches/0007-Use-std-atomics-and-std-mutex-for-portability.patch
+      ./patches/0008-Add-Meson-build-system.patch
+      ./patches/0009-Add-CrashReportedClient-header.patch
+      ./patches/0010-Provide-mach-compatibility-headers-based-on-LLVM-s-h.patch
+      ./patches/0011-Support-LTO-in-nixpkgs.patch
+      ./patches/0012-Add-vendored-libtapi-to-the-ld64-build.patch
+      ./patches/0013-Modify-vendored-libtapi-to-build-with-upstream-LLVM.patch
+      ./patches/0014-Move-libcodedirectory-to-its-own-subproject.patch
+      ./patches/0015-Set-the-version-string-in-the-build.patch
+      ./patches/0016-Replace-corecrypto-and-CommonCrypto-with-OpenSSL.patch
+      ./patches/0017-Add-libcd_is_blob_a_linker_signature-implementation.patch
+      ./patches/0018-Add-dyldinfo-to-the-ld64-build.patch
+      ./patches/0019-Fix-dyldinfo-build.patch
+      ./patches/0020-Use-STL-containers-instead-of-LLVM-containers.patch
+    ]
+    ++ lib.optionals (hasEmbed) [
+      ./patches/0004-Add-compile_stubs.h-using-Clang-s-embed-extension-fo.patch
+    ];
+
+  prePatch = ''
+    # Copy dyldinfo source files
+    cp ${ld64src}/doc/man/man1/dyldinfo.1 doc/man/man1/dyldinfo.1
+    cp ${ld64src}/src/other/dyldinfo.cpp src/other/dyldinfo.cpp
+
+    # Copy files needed from libtapi by ld64
+    mkdir -p subprojects/libtapi/tapi
+    cp ${libtapisrc}/tools/libtapi/*.cpp subprojects/libtapi
+    cp ${libtapisrc}/LICENSE.TXT subprojects/libtapi/LICENSE.TXT
+
+    declare -a tapiHeaders=(
+      APIVersion.h
+      Defines.h
+      LinkerInterfaceFile.h
+      PackedVersion32.h
+      Symbol.h
+      Version.h
+      Version.inc.in
+      tapi.h
+    )
+    for header in "''${tapiHeaders[@]}"; do
+      cp ${libtapisrc}/include/tapi/$header subprojects/libtapi/tapi/$header
+    done
```

Comment:
Wouldn’t it be simpler to just `ln -s ${libtapisrc} subprojects/libtapi`? The extra files won’t hurt anything.

Reply:
That would be a store path, which I can’t modify.

Reply:
Fair enough, but just `cp`ing the whole tree would be simpler and have very little cost, right?

Reply:
My preference would be not to drop the patch adding the vendored libtapi because copying tightly couples ld64 to version of libtapi in nixpkgs. If I can’t do that, I’d prefer to do what that patch is doing (adding these files).

Reply:
Alright. I personally think it would be simpler to just copy over the whole tree, but I don’t object to the way it’s done now. (I would probably have just `git subtree`d the entirety of `libtapi` into the repository too, anyway, so…)

FWIW I would also be fine with pinning an exact `libtapi` commit in the file.